src/20240528_Multi_GHOCrossChainLaunch/AaveV3Arbitrum_GHOCrossChainLaunch_20240528.sol

```diff
@@ -0,0 +1,308 @@
+import {GovernanceV3Arbitrum} from 'aave-address-book/GovernanceV3Arbitrum.sol';
+import {MiscArbitrum} from 'aave-address-book/MiscArbitrum.sol';
+import {AaveV3PayloadArbitrum} from 'aave-helpers/v3-config-engine/AaveV3PayloadArbitrum.sol';
+import {EngineFlags} from 'aave-helpers/v3-config-engine/EngineFlags.sol';
+import {IAaveV3ConfigEngine} from 'aave-helpers/v3-config-engine/IAaveV3ConfigEngine.sol';
+import {IV3RateStrategyFactory} from 'aave-helpers/v3-config-engine/IV3RateStrategyFactory.sol';
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {SafeERC20} from 'solidity-utils/contracts/oz-common/SafeERC20.sol';
+import {ICreate3Factory} from 'solidity-utils/contracts/create3/interfaces/ICreate3Factory.sol';
+import {TransparentUpgradeableProxy} from 'solidity-utils/contracts/transparent-proxy/TransparentUpgradeableProxy.sol';
+import {UpgradeableBurnMintTokenPool} from 'ccip/v0.8/ccip/pools/GHO/UpgradeableBurnMintTokenPool.sol';
+import {UpgradeableTokenPool} from 'ccip/v0.8/ccip/pools/GHO/UpgradeableTokenPool.sol';
+import {RateLimiter} from 'ccip/v0.8/ccip/libraries/RateLimiter.sol';
+import {UpgradeableGhoToken} from 'gho-core/gho/UpgradeableGhoToken.sol';
+import {IGhoToken} from 'gho-core/gho/interfaces/IGhoToken.sol';
+
+library Utils {
+  bytes32 public constant GHO_IMPL_DEPLOY_SALT = bytes32(uint256(keccak256('gho.token.impl')));
+  bytes32 public constant GHO_DEPLOY_SALT = bytes32(uint256(keccak256('gho.token')));
+  uint256 public constant GHO_SEED_AMOUNT = 1e18;
+  bytes32 public constant CCIP_TOKEN_POOL_IMPL_DEPLOY_SALT =
+    bytes32(uint256(keccak256('gho.ccip.impl')));
+  bytes32 public constant CCIP_TOKEN_POOL_DEPLOY_SALT = bytes32(uint256(keccak256('gho.ccip')));
+  address public constant CCIP_ARM_PROXY = 0xC311a21e6fEf769344EB1515588B9d535662a145;
+  address public constant CCIP_ROUTER = 0x141fa059441E0ca23ce184B6A78bafD2A517DdE8;
+  uint256 public constant CCIP_BUCKET_CAPACITY = 1_000_000e18; // 1M
+  uint64 public constant CCIP_ETH_CHAIN_SELECTOR = 5009297550715157269;
+
+  function deployGhoToken() internal returns (address imple, address proxy) {
+    // Deploy imple
+    imple = ICreate3Factory(MiscArbitrum.CREATE_3_FACTORY).create(
+      GHO_IMPL_DEPLOY_SALT,
+      type(UpgradeableGhoToken).creationCode
+    );
+
+    // proxy deploy and init
+    bytes memory ghoTokenInitParams = abi.encodeWithSignature(
+      'initialize(address)',
+      GovernanceV3Arbitrum.EXECUTOR_LVL_1 // owner
+    );
+    bytes memory creationCode = abi.encodePacked(
+      type(TransparentUpgradeableProxy).creationCode,
+      abi.encode(
+        imple,
+        MiscArbitrum.PROXY_ADMIN, // proxy admin
+        ghoTokenInitParams
+      )
+    );
+    proxy = ICreate3Factory(MiscArbitrum.CREATE_3_FACTORY).create(GHO_DEPLOY_SALT, creationCode);
+  }
+
+  function deployCcipTokenPool(address ghoToken) external returns (address imple, address proxy) {
```

Comment:
Doesn't this potentially make the whole execution revert?

Reply:
I believe this is safe given that create3 address depends on the executor's signer